src/kubernetes.cc

```diff
@@ -568,70 +569,111 @@ json::value KubernetesReader::QueryMaster(const std::string& path) const
 
 namespace {
 struct Watcher {
-  Watcher(std::function<void(json::value)> event_callback,
+  Watcher(const std::string& endpoint,
+          std::function<void(json::value)> event_callback,
           std::unique_lock<std::mutex>&& completion, bool verbose)
-      : completion_(std::move(completion)), event_callback_(event_callback),
-        remaining_chunk_bytes_(0), verbose_(verbose) {}
+      : name_("Watcher(" + endpoint + ")"),
+        completion_(std::move(completion)), event_callback_(event_callback),
+        remaining_chunk_bytes_(0), verbose_(verbose), exception_() {}
   ~Watcher() {}  // Unlocks the completion_ lock.
+
+ private:
+  // A representation of all watch-related errors.
+  class WatcherException {
+   public:
+    WatcherException(const std::string& what) : explanation_(what) {}
+    const std::string& what() const { return explanation_; }
+   private:
+    std::string explanation_;
```

Comment:
Why is this called `explanation_` instead of `what_`?

Reply:
I'm fine with `e.what()`, I just got confused why the local variable is called `explanation_`, not critical in either case so happy to just approve.